src/Spice86.Core/Emulator/Memory/DmaChannel.cs

```diff
@@ -20,9 +22,12 @@ public sealed class DmaChannel {
     private int _transferRate;
     private readonly Stopwatch _transferTimer = new();
     private readonly IMemory _memory;
+    private readonly State _state;
+    private long? _signalCompletionAfter;
 
-    internal DmaChannel(IMemory memory) {
+    internal DmaChannel(IMemory memory, State state) {
```

Comment:
The DmaController class should pass the _state reference in order for the build to pass.